contracts/bindings/scripts/commenttypes.go

```diff
@@ -0,0 +1,130 @@
+// commenttypes comments out type definitions in a Go file
+// Usage: go run commenttypes.go -- <file> <type1> <type2> ...
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/omni-network/omni/lib/errors"
+)
+
+func main() {
+	err := run()
+
+	if err != nil {
+		fmt.Printf("commenttypes failed: %v\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Println("commenttypes succeeded")
+}
+
+func run() error {
+	if len(os.Args) < 3 {
+		return errors.New("usage: go run commenttypes.go -- <file> <type1> <type2> ...")
+	}
+
+	// os.Args[0] is the name of the command
+	// os.Args[1] is either <file> or  "--" separator
+
+	firstArgIdx := 1
+	if os.Args[1] == "--" {
+		firstArgIdx = 2
+	}
+
+	filepath := os.Args[firstArgIdx]
+	types := os.Args[firstArgIdx+1:]
+
+	if len(types) == 0 {
+		return errors.New("no types")
+	}
+
+	inputF, err := os.Open(filepath)
+	if err != nil {
+		return errors.Wrap(err, "open file")
+	}
+	defer inputF.Close()
+
+	scanner := bufio.NewScanner(inputF)
+	var content string
+	for scanner.Scan() {
+		content += scanner.Text() + "\n"
+	}
+
+	modified := commentTypes(content, types)
+
+	err = os.WriteFile(filepath, []byte(modified), 0600)
+	if err != nil {
+		return errors.Wrap(err, "write file")
+	}
+
+	return nil
+}
+
+// commentTypes comments out type definitions in a Go file.
+func commentTypes(content string, types []string) string {
+	var modifiedLines []string
+	lines := strings.Split(content, "\n")
+
+	// tracks whether we're in a type definition
+	//
+	// type Name {
+	// ...
+	// }
```

Comment:
Ideally, we need to do this using some form of AST.

Reply:
yeah but this is pretty lowstakes, single usecase, so quick and dirty is fine

Reply:
Take a look at [this library](https://github.com/unpackdev/solgo).
Alternatively [this SO answer](https://ethereum.stackexchange.com/questions/28523/how-to-go-from-solidity-ast-back-to-source-code) recommends [this library](https://github.com/chao-peng/SIF).

Reply:
I didn't realize that this was Go code. Here is [an example](https://go.dev/src/go/ast/example_test.go).